debugging/coreclr/dockerDebugConfigurationProvider.ts

```diff
@@ -0,0 +1,250 @@
+import { DockerManager, LaunchResult } from './dockerManager';
+import { FileSystemProvider } from './fsProvider';
+import { NetCoreProjectProvider } from './netCoreProjectProvider';
+import { OSProvider } from './osProvider';
+import { Prerequisite } from './prereqManager';
+
+interface DockerDebugBuildOptions {
+    context?: string;
+    dockerfile?: string;
+    tag?: string;
+    target?: string;
+}
+
+interface DockerDebugRunOptions {
+    containerName?: string;
+    os?: PlatformOS;
+}
+
+interface DebugConfigurationBrowserBaseOptions {
+    enabled?: boolean;
+    command?: string;
+    args?: string;
+}
+
+interface DebugConfigurationBrowserOptions extends DebugConfigurationBrowserBaseOptions {
+    windows?: DebugConfigurationBrowserBaseOptions;
+    osx?: DebugConfigurationBrowserBaseOptions;
+    linux?: DebugConfigurationBrowserBaseOptions;
+}
+
+interface DockerDebugConfiguration extends DebugConfiguration {
+    appFolder?: string;
+    appOutput?: string;
+    appProject?: string;
+    dockerBuild?: DockerDebugBuildOptions;
+    dockerRun?: DockerDebugRunOptions;
+}
+
+export class DockerDebugConfigurationProvider implements DebugConfigurationProvider {
+    constructor(
+        private readonly debugSessionManager: DebugSessionManager,
+        private readonly dockerManager: DockerManager,
+        private readonly fsProvider: FileSystemProvider,
+        private readonly osProvider: OSProvider,
+        private readonly netCoreProjectProvider: NetCoreProjectProvider,
+        private readonly prerequisite: Prerequisite) {
+    }
+
+    public provideDebugConfigurations(folder: WorkspaceFolder | undefined, token?: CancellationToken): ProviderResult<DebugConfiguration[]> {
+        return [
+            {
+                name: 'Docker: Launch .NET Core ASP.NET (Preview)',
+                type: 'docker-coreclr',
+                request: 'launch',
+                preLaunchTask: 'build',
+                dockerBuild: {
+                },
+                dockerRun: {
+                }
+            }
+        ];
+    }
+
+    public resolveDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration, token?: CancellationToken): ProviderResult<DebugConfiguration | undefined> {
+        return this.resolveDockerDebugConfiguration(folder, debugConfiguration);
+    }
+
+    private static resolveFolderPath(folderPath: string, folder: WorkspaceFolder): string {
+        return folderPath.replace(/\$\{workspaceFolder\}/g, folder.uri.fsPath);
+    }
+
+    private async resolveDockerDebugConfiguration(folder: WorkspaceFolder | undefined, debugConfiguration: DockerDebugConfiguration): Promise<DebugConfiguration | undefined> {
+        if (!folder) {
+            return undefined;
+        }
+
+        const prerequisiteSatisfied = await this.prerequisite.checkPrerequisite();
+
+        if (!prerequisiteSatisfied) {
+            return undefined;
```

Comment:
should this throw?

Reply:
In my opinion no, as the prerequisite checks themselves already displaying their corresponding actionable messages.  If we throw here, then VS Code would display another error message which would be less actionable as well as tend to dilute the prerequisite check errors.